cmake/templates/CMakeLists.txt.in

```diff
@@ -0,0 +1,19 @@
+cmake_minimum_required(VERSION 3.2.2)
+
+# This file is copied into the build directory in order to build googletest
+# See the guidelines at the URL below for a more detailed explanation
+# https://github.com/google/googletest/blob/master/googletest/README.md
+
+project(googletest-download NONE)
+
+include(ExternalProject)
+ExternalProject_Add(googletest
+    GIT_REPOSITORY https://github.com/google/googletest.git
+    GIT_TAG master
+    SOURCE_DIR "${CMAKE_CURRENT_BINARY_DIR}/googletest-src"
```

Comment:
It seems strange to me that a template would be needed for this

Reply:
I agree - however, it's the way that googletests suggests that one should integrate it, so I'm not going to argue with them. 

(see: https://github.com/google/googletest/blob/master/googletest/README.md)